### PR TITLE
Backup .sra Files Before fasterq-dump

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

https://github.com/AlexsLemonade/refinebio/issues/930

## Methods

This change is isolated from this branch: https://github.com/AlexsLemonade/refinebio/pull/932/files

but that changeset can't land until Nomad 0.8.7+ is released, which won't be for another ~2(+?) weeks.

### CODE REVIEW

File: workers/data_refinery_workers/processors/salmon.py [hunk 1]
```diff
@@ -60,11 +60,39 @@ def _prepare_files(job_context: Dict) -> Dict:
...
+        logger.error("Was told to process a non-existent file - why did this happen?",
+            input_file_path=job_context["input_file_path"],
+            processor_job=job_context["job_id"]
+        )
+        job_context["job"].failure_reason = "Missing input file: " + str(job_context["input_file_path"])
+        job_context["success"] = False
+        return job_context
+
+    # Copy the .sra file so fasterq-dump can't corrupt it.
+    if job_context["input_file_path"][-4:] == ".sra":
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/salmon.py]: Can these ever be `.SRA`?

File: workers/data_refinery_workers/processors/salmon.py [hunk 2]
```diff
@@ -60,11 +60,39 @@ def _prepare_files(job_context: Dict) -> Dict:
...
+        new_input_file_path = os.path.join(job_context["work_dir"], original_files[0].filename)
+        shutil.copyfile(job_context["input_file_path"], new_input_file_path)
+        job_context['input_file_path'] = new_input_file_path
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/salmon.py]: You also need to copy `input_file_path_2` over for the same reasons right?

File: workers/data_refinery_workers/processors/salmon.py (hunk 1 above)
Comment (Miserlou) [reply workers/data_refinery_workers/processors/salmon.py]: In practice no but we can check

File: workers/data_refinery_workers/processors/salmon.py (hunk 2 above)
Comment (Miserlou) [reply workers/data_refinery_workers/processors/salmon.py]: fasterq-dump is only ever going to take one file at a time, but we might as well copy both

Comment (kurtwheeler) [reply workers/data_refinery_workers/processors/salmon.py]: It's  not a might as well thing is it, fasterq-dump will be called on both `input_file_path` and `input_file_path_2` so we need to back them both up right?

File: workers/data_refinery_workers/processors/salmon.py [hunk 3]
```diff
@@ -87,12 +84,28 @@ def _prepare_files(job_context: Dict) -> Dict:
         job_context["success"] = False
         return job_context
 
+    if len(original_files) == 2:
+        job_context["input_file_path_2"] = original_files[1].absolute_file_path
+    if not os.path.exists(job_context["input_file_path_2"]):
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/salmon.py]: This if statement is assuming that the job_context has had the `input_file_path_2` key set, but it is valid for that to not be populated. This if should just start with a `job_context.get("input_file_path_2", False) and` like you have below.

Comment (kurtwheeler) [reply workers/data_refinery_workers/processors/salmon.py]: Or it could just be nested within the `if len(original_files) == 2` conditional block.

**Pull request merged**